day01/index.js

```diff
@@ -0,0 +1,43 @@
+const R = require('ramda');
+const fs = require('fs');
+const path = require('path');
+
+const inputData = fs.readFileSync(path.resolve(__dirname, 'data.txt'), 'utf8');
+
+const convertInputToArr = input =>
+  input
+    .toString()
+    .split(/\n|\$/)
+    .map(item => parseInt(item, 10));
+
+const getFuelRequirementByMass = mass => mass.map(m => Math.floor(m / 3) - 2);
+
+const reduceFuelArr = fuelArr =>
+  fuelArr.reduce((total, currentValue) => total + currentValue);
+
+const getFuelForMass = R.pipe(
+  convertInputToArr,
+  getFuelRequirementByMass,
+  reduceFuelArr
+);
+
+//-------------------------------------------------------------------
+// PART TWO
+//-------------------------------------------------------------------
+const getFuelForFuelWeight = totalFuel => {
+  if (totalFuel <= 0) {
+    return totalFuel;
+  } else {
+    const newValue = getFuelForMass(totalFuel);
+
+    return newValue < 0
+      ? totalFuel
+      : (totalFuel += getFuelForFuelWeight(newValue));
+  }
+};
```

Comment:
This is soooooo close! I like the recursive call but i really think there is one ternary in there too many and could be refactored a bit. Also, stretch goal, that recursive call is not tail call optimized and could fail for large data sets. Check out a discussion of tail call optimization in the Allonge book in the Sparkbox bookshelf to see what I mean.

Reply:
This is real page 99 and pdf page 122 (tail-call optimization)

Reply:
Thanks for the info @gtempus . This has been implemented in ceacc2d5283a3882166f2c9c022109815a0cee8e